### PR TITLE
Implemented updated auth SDK

### PR DESCRIPTION
**Pull request by shrinathprabhu**
Resolves [AR-1400]().

## Changes

1. Moved Arcana Auth logic from `services` to `use/arcanaAuth`
2. Replaced static login types with `SocialLoginType` constants
3. Updated the auth with `AuthProvider.init` method
4. Moved Auth Config to `env`
5. Replaced popup flow with redirect flow
6. Removed `SSORedirect.vue` as it is not required in redirect flow
7. Removed unused code from `router.js`
8. Rendered `router-view` only if Arcana Auth is initialized

## Checklist

- [x] The branch name follows the format: `developer/AR-XXX-issue-name`.
- [x] The changes have been tested locally.

### CODE REVIEW

File: src/pages/AppLoginV2.vue [hunk 1]
```diff
@@ -205,10 +202,11 @@ export default {
         loading.value = true;
         if (!isLoggedIn) {
           loadingMessage.value = "Signing In...";
-          await arcanaAuth.loginWithSocial(type);
+          await loginWithSocial(type);
         }
         loadingMessage.value = "Fetching user info...";
-        const { userInfo, privateKey } = arcanaAuth.getUserInfo();
+        const { userInfo, privateKey } = await fetchUserDetails();
+        console.log({ userInfo, privateKey });
```
Comment (ajithranka) [inline src/pages/AppLoginV2.vue:209]: Stray console.log.

File: src/use/arcanaAuth.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  async function init() {
+    console.log("Before Init");
+    if (!authInstance) {
+      authInstance = await AuthProvider.init({
+        appId: ARCANA_APP_ID,
+        network: ARCANA_AUTH_NETWORK,
+        flow: ARCANA_AUTH_FLOW,
+      });
+    }
+    console.log("After Init");
```
Comment (ajithranka) [inline src/use/arcanaAuth.js:22]: Stray console.log.

**Review by ajithranka — CHANGES_REQUESTED** [review]
Question: Why did we switch from the popup flow to the redirect flow?

File: src/pages/AppLoginV2.vue (hunk 1 above)
Comment (shrinathprabhu) [reply src/pages/AppLoginV2.vue:209]: Yeah

**Comment by shrinathprabhu** [comment]
> Question: Why did we switch from the popup flow to the redirect flow? If we are allowing configuring the flow type in the environment variable, we should keep the SSORedirect component.

1. We required redirect flow to support mobile browsers and older version of Safari. In those popup mode doesn't work, so we implemented redirect mode in sdk. So if anyone wants to develop apps for mobile browsers, they can use redirect mode. We want our dashboard to be supported in mobile browsers as of now. We can discuss this with the team
2. Even if we change flow type from `.env`, it'll still require SSORedirect code to be changed, since in `popup` flow we use `AuthProvider.handleRedirect()` in SSORedirect and in `redirect` flow we directly use `authInstance.fetchUserInfo()`. Changing mode will still require code changes and won't be handled out of the box even if we keep `SSORedirect.vue`. For `redirect` flow we don't actually need SSORedirect as the entire thing will be handled in login page itself

**Comment by ajithranka** [comment]
> > Question: Why did we switch from the popup flow to the redirect flow? If we are allowing configuring the flow type in the environment variable, we should keep the SSORedirect component.
> 
> 1. We required redirect flow to support mobile browsers and older version of Safari. In those popup mode doesn't work, so we implemented redirect mode in sdk. So if anyone wants to develop apps for mobile browsers, they can use redirect mode. We want our dashboard to be supported in mobile browsers as of now. We can discuss this with the team
> 2. Even if we change flow type from `.env`, it'll still require SSORedirect code to be changed, since in `popup` flow we use `AuthProvider.handleRedirect()` in SSORedirect and in `redirect` flow we directly use `authInstance.fetchUserInfo()`. Changing mode will still require code changes and won't be handled out of the box even if we keep `SSORedirect.vue`. For `redirect` flow we don't actually need SSORedirect as the entire thing will be handled in login page itself

Makes sense, @shrinathprabhu. Thanks for the explanation. 

Should we remove the `VITE_ARCANA_AUTH_FLOW` variable from `.env` and set it to "redirect" in the code?

**Comment by shrinathprabhu** [comment]
> > > Question: Why did we switch from the popup flow to the redirect flow? If we are allowing configuring the flow type in the environment variable, we should keep the SSORedirect component.
> > 
> > 
> > 
> > 1. We required redirect flow to support mobile browsers and older version of Safari. In those popup mode doesn't work, so we implemented redirect mode in sdk. So if anyone wants to develop apps for mobile browsers, they can use redirect mode. We want our dashboard to be supported in mobile browsers as of now. We can discuss this with the team
> > 2. Even if we change flow type from `.env`, it'll still require SSORedirect code to be changed, since in `popup` flow we use `AuthProvider.handleRedirect()` in SSORedirect and in `redirect` flow we directly use `authInstance.fetchUserInfo()`. Changing mode will still require code changes and won't be handled out of the box even if we keep `SSORedirect.vue`. For `redirect` flow we don't actually need SSORedirect as the entire thing will be handled in login page itself
> 
> Makes sense, @shrinathprabhu. Thanks for the explanation.
> 
> Should we remove the `VITE_ARCANA_AUTH_FLOW` variable from `.env` and set it to "redirect" in the code?

Yeah we can

**Review by ajithranka — APPROVED** [review]
Looks good, thanks @shrinathprabhu.

**Pull request merged**